tools/perf-report.cc

```diff
@@ -238,6 +222,10 @@ print_entry(Addr2line &addr2line, uint64_t count, uint64_t total,
   printf("\n");
 }
 
+#ifdef __APPLE__
+#define pipe2(a,b) pipe(a)
```

Comment:
This ignores the flags (ie O_CLOEXEC) passed to the function. Do you know whether that matters in the code below? If not, it might make sense just to replace pipe2 -> pipe and not have this special define

Reply:
I'm not sure exactly what this code does, however based on [this](http://man7.org/linux/man-pages/man2/open.2.html) I think it is safe to replace with `fcntl` since I don't think the race condition doesn't apply to pipes.